### PR TITLE
Use a local wrapper for Jitsi calls

### PR DESCRIPTION
**Pull request by turt2live**
**Requires https://github.com/vector-im/riot-web/pull/12780**

See required PR for more information.

### CODE REVIEW

File: docs/jitsi.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Jitsi Wrapper
+
+**Note**: these are developer docs. Please consult your client's documentation for
```
Comment (jryans) [inline docs/jitsi.md]:
```suggestion
**Note**: These are developer docs. Please consult your client's documentation for
```

File: docs/jitsi.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+instructions on setting up Jitsi.
+
+The react-sdk wraps all Jitsi call widgets in a local wrapper called `jitsi.html`
+which takes several parameters:
+
+*Query string*:
+* `widgetId`: The ID of the widget. This is needed for communication back to the 
+  react-sdk.
+* `parentUrl`: The URL of the parent window. This is also ideally needed for
```
Comment (jryans) [inline docs/jitsi.md]: "ideally needed"...? What does that mean here?

```suggestion
* `parentUrl`: The URL of the parent window. This is also needed for
```

File: src/utils/WidgetUtils.js [hunk 3]
```diff
@@ -468,4 +473,31 @@ export default class WidgetUtils {
 
         return encodeURIComponent(`${widgetLocation}::${widgetUrl}`);
     }
+
+    static getLocalJitsiWrapperUrl(forLocalRender = false) {
```
Comment (jryans) [inline src/utils/WidgetUtils.js]: Maybe use options object to force naming the boolean at call sites?

File: src/widgets/WidgetApi.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        this.readyPromise = new Promise<any>(resolve => this.readyPromiseResolve = resolve);
+
+        window.addEventListener("message", event => {
+            if (event.origin !== this.origin) return; // ignore: invalid origin
+            if (!event.data) return; // invalid schema
+            if (event.data.widgetId !== this.widgetId) return; // not for us
+
+            const payload = <WidgetRequest>event.data;
+            if (payload.api === WidgetApiType.ToWidget && payload.action) {
+                console.log(`[WidgetAPI] Got request: ${JSON.stringify(payload)}`);
```
Comment (jryans) [inline src/widgets/WidgetApi.ts]: Are these logs safe to keep in?

File: docs/jitsi.md (hunk 2 above)
Comment (turt2live) [reply docs/jitsi.md]: Developers can choose to not use this at risk of security vulnerabilities and noisy Chrome extensions.

File: src/widgets/WidgetApi.ts (hunk 4 above)
Comment (turt2live) [reply src/widgets/WidgetApi.ts]: Yes, they're already made by existing managers and Riot.

**Pull request merged**